api/endpoints.md

```diff
@@ -0,0 +1,181 @@
+# User Resource
+
+## `GET /users`
+
+Read users.
+
+### Authorization
+
+-   `read:users` — Student, Reviewer, Interviewer, Admin
+
+### Query Parameters
+
+-   `?role=reviewer`
+-   `?id=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX`
+
+## `POST /users`
+
+Create a new user and give them the User role in Auth0.
+
+### Authorization
+
+This route will only be protected by JWT authentication.
+
+# Role Resource
+
+## `GET /users/{user}/roles`
+
+Read all roles for a user.
+
+### Authorization
+
+-   `read:roles` — Student, Reviewer, Interviewer, Admin
+
+## `PUT /users/{user}/roles/{role}`
+
+Give a user a new role.
+
+### Authorization
+
+-   `create:roles` — Admin
+
+## `DELETE /users/{user}/roles/{role}`
+
+Revoke a users role.
+
+### Authorization
+
+-   `delete:roles` — Admin
+
+# Resume Review Resource
+
+## `GET /resume-reviews`
+
+Read resume reviews.
+
+### Authorization
+
+-   `read_my:resume_reviews` — Student
+-   `read_all:resume_reviews` — Reviewer, Admin
+
+### Query Parameters
+
+-   `?reviewer=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX`
+-   `?reviewee=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX`
+-   `?state=created`
+-   `?id=XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX`
+
+## `POST /resume-reviews`
+
+Create a new resume review.
+
+### Authorization
+
+-   `create:resume_reviews` — Student, Admin
+
+## `PATCH /resume-reviews/{resume-review}`
+
+Update a resume review.
```

Comment:
Can you remind me what this updates?

Reply:
You would use this to update the `state` of a resume review for example e.g. a resume review is finished and you need to mark it as `finished`.